src/coap_debug.c

```diff
@@ -87,9 +87,8 @@ print_timestamp(char *s, size_t len, coap_tick_t t) {
 COAP_STATIC_INLINE size_t
 print_timestamp(char *s, size_t len, coap_tick_t t) {
 #ifdef HAVE_SNPRINTF
-  return snprintf(s, len, "%u.%03u",
-                  (unsigned int)coap_ticks_to_rt(t),
-                  (unsigned int)(t % COAP_TICKS_PER_SECOND));
+  return snprintf(s, len, "%u",
```

Comment:
This gives us time in seconds whereas this PR says we want to have microseconds. `coap_ticks_to_rt_us()` probably is what we want here, preferably in a more readable format "%u.%06u".

Reply:
This intentionally is in seconds (so both variants of print_timestamp() return the same thing) - static print_timestamp()'s caller appends the microseconds. At one level, it makes sense that print_timestamp() returns also microsecs modified in both cases.